src/@types/matrix-sdk-crypto-wasm.d.ts

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2024 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import type * as RustSdkCryptoJs from "@matrix-org/matrix-sdk-crypto-wasm";
+
+declare module "@matrix-org/matrix-sdk-crypto-wasm" {
+    interface OlmMachine {
```

Comment:
Why do we need this? isn't it already in the index.d.ts of `matrix-sdk-crypto-wasm`?

Reply:
Yes but with return types of `any` - see https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/issues/110

Reply:
![image](https://github.com/matrix-org/matrix-js-sdk/assets/2403652/a7c69172-a706-446b-a77b-d36786222497)

Reply:
This should be fixed on the wasm side; https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/pull/123 does so but it will need a release cycle :/

Reply:
Well, that PR doesn't help with the return type of `SecretsBundle.to_json`, to be fair. On the other hand: do you actually need to introspect the result of `SecretsBundle.to_json` outside of tests (where I would be inclined to do some `@ts-ignore`ing rather than have a separate definition to keep in step)?

Reply:
> Well, that PR doesn't help with the return type of SecretsBundle.to_json, to be fair. On the other hand: do you actually need to introspect the result of SecretsBundle.to_json outside of tests (where I would be inclined to do some @ts-ignoreing rather than have a separate definition to keep in step)?

It should still be a useful type otherwise you'll get junk into the import method. If it isn't meant to be introspectable then it should be a branded type so a round-trip export->import works without needing to rely on `any`